### PR TITLE
All the work from the DriveSystem Group

### PR DESCRIPTION
**Pull request by Sam-Belliveau**
This is pretty much done, we just need to get the ports and calculate values, which we can't do until we have a drive train

### CODE REVIEW

**Comment by Sam-Belliveau** [comment]
Here is where the newbies fill in the gaps, I just provided a (fairly large sorry) template with all the complicated stuff

File: src/main/java/com/stuypulse/robot/subsystems/Drivetrain.java [hunk 1]
```diff
@@ -1,60 +1,300 @@
...
+            motor.setInverted(inverted);
+        }
+
+        for (CANSparkMax motor : mRightMotors) {
+            motor.setInverted(inverted);
+        }
+    }
+
+    /**
+     * Sets drivetrain into high gear
```
Comment (PrathamRawat) [inline src/main/java/com/stuypulse/robot/subsystems/Drivetrain.java]: Low

File: src/main/java/com/stuypulse/robot/subsystems/Drivetrain.java [hunk 2]
```diff
@@ -1,60 +1,300 @@
...
+     */
+    public void setLowGear() {
+        stop();
+        mGear = DrivetrainGear.LOW;
+        mGearShift.set(false);
+    }
+
+    /**
+     * Sets drivetrain into low gear
```
Comment (PrathamRawat) [inline src/main/java/com/stuypulse/robot/subsystems/Drivetrain.java:159]: High

**Review by PrathamRawat — CHANGES_REQUESTED** [review]
A few things need to be changed
Please use standard Java naming conventions instead of the m / k convention (the big one)
I have qualms merging this in unless Tahsin and Kevin both also give their approval of this structure
I'll do a deeper reading of the code structure once that is done

**Comment by PrathamRawat** [comment]
Merge master into your branch and fix merge conflicts

**Review by PrathamRawat — APPROVED** [review]

**Pull request merged**